aea/registries/base.py

```diff
@@ -518,7 +518,9 @@ def __init__(self, directory: Optional[Union[str, os.PathLike]] = None):
         self.protocol_registry = ProtocolRegistry()
         self.handler_registry = HandlerRegistry()
         self.behaviour_registry = ComponentRegistry[Behaviour]()
-        self.task_registry = ComponentRegistry[Task]()
+        self.task_registry = ComponentRegistry[
```

Comment:
Unless we use it to keep track of the initialized tasks. But that doesn't really make sense as it limits the programmatic usage. So I think it can be removed.